### DISCUSSION TITLE
Difference between v2 and v3

### DISCUSSION THREAD

**Discussion by kirkpabk** [microsoft/azure-pipelines-agent#4099] [General]
I sort of feel ignorant for asking and missing the thread, but what are the primary architectural differences between v2 and v3?

**Comment by hackup86** [comment]
if im not wrong v3 is for newer systems as it builds on .net6 and openssl without deprecated features

**Comment by ferben** [comment]
I see also build for macARM64.
So I'd like to know how can I update from x86 emulated agent to native ARM agent on my M1 computer?
Will update process automatically recognize processor and update to native binary?

**Reply by ferben** [reply]
> I sort of feel ignorant for asking and missing the thread, but what are the primary architectural differences between v2 and v3?

https://aka.ms/azdo-pipeline-agent-version

**Comment by ferben** [comment]
Ooops, update to 3.218.0 is available for me now with standard update from devops webUI.
Unfortunately macOS still keep x86_64 binaries even on M1 mac → manually updated to native arm64 version.
